File: contract/src/main/java/rrlane/leasing/contract/service/LeasingContractService.java
package rrlane.leasing.contract.service;

import rrlane.leasing.contract.dto.CustomerDTO;
import rrlane.leasing.contract.dto.LeasingContractDTO;
import rrlane.leasing.contract.entity.LeasingContract;
import rrlane.leasing.entity.Vehicle;

import java.util.List;

public interface LeasingContractService {
    public String saveLeasingContract(LeasingContractDTO contractDTO);

    public CustomerDTO getCustomerByName(String customerName);

    public Vehicle getVehicleByDetails(String vehicleDetails);//TODO to use or to remove?

    public List<LeasingContract> view();
}

File: contractapi/src/main/kotlin/de/leasing/contract/api/service/LeasingContractService.kt
package de.leasing.contract.api.service

import de.leasing.contract.api.entity.dto.LeasingContractReq
import de.leasing.contract.api.entity.dto.LeasingContractResp

interface LeasingContractService {
    fun saveLeasingContract(contractReq: LeasingContractReq): LeasingContractResp
    fun listAllLeasingContracts(page: Int, size: Int): Iterable<LeasingContractResp>
    fun listAllLeasingContracts(sortBy: String?): Iterable<LeasingContractResp>
    fun getLeasingContractById(id: String): LeasingContractResp
}
